proxy/mux/mux_test.go

```diff
@@ -370,6 +382,286 @@ func (s *ServerSuite) TestHostKeyPairUpdate(c *C) {
 	c.Assert(GETResponse(c, b.FrontendURL("/")), Equals, "Hi, I'm endpoint")
 }
 
+func (s *ServerSuite) TestServerHTTPSAutoCert(c *C) {
+	var req *http.Request
+	e := testutils.NewHandler(func(w http.ResponseWriter, r *http.Request) {
+		req = r
+		w.Write([]byte("hi https"))
+	})
+	defer e.Close()
+
+	man := &autocert.Manager{
+		Prompt: autocert.AcceptTOS,
+	}
+	url, finish := startACMEServerStub(c, man, "example.org")
```

Comment:
I struggled with this one a lot. The problem with the staging environment is that you still need to be behind a real host with an internet-facing name to generate the cert. So it works for integration tests, but not Unit Tests. What I suggest is we keep the UTs in place, since they validate the basic logic, and then add integration tests as a followup to test real-world usage.

Reply:
You are right, I did not thought